cpp/src/arrow/gpu/cuda_memory.cc

```diff
@@ -501,5 +502,27 @@ Result<std::shared_ptr<MemoryManager>> DefaultMemoryMapper(ArrowDeviceType devic
   }
 }
 
+Result<std::shared_ptr<MemoryManager>> DefaultGPUMemoryMapper(int64_t device_id) {
+  ARROW_ASSIGN_OR_RAISE(auto device, arrow::cuda::CudaDevice::Make(device_id));
+  return device->default_memory_manager();
+}
+
+Status RegisterCUDADeviceInternal() {
+  RETURN_NOT_OK(
+      RegisterDeviceMemoryManager(DeviceAllocationType::kCUDA, DefaultGPUMemoryMapper));
+  RETURN_NOT_OK(RegisterDeviceMemoryManager(DeviceAllocationType::kCUDA_HOST,
+                                            DefaultGPUMemoryMapper));
+  RETURN_NOT_OK(RegisterDeviceMemoryManager(DeviceAllocationType::kCUDA_MANAGED,
+                                            DefaultGPUMemoryMapper));
+  return Status::OK();
+}
+
+std::once_flag cuda_registered;
+
+Status RegisterCUDADevice() {
```

Comment:
Is there a reason for not doing this automatically?

Reply:
Well, if they have a need to a dedicated CUDA mapper, they can just pass their own `DeviceMemoryMapper` when importing, AFAICT. What do you think @zeroshade ?

Reply:
That's true. I was going to suggest that with this registration mechanism, we don't necessarily need to keep the device mapper keyword, but that's actually a reason to keep it

Reply:
I pushed a commit that registers the CUDA device by default and therefore removes the public `RegisterCudaDevice` function.